File: frontend/lib/config/request.dart
import 'dart:convert';
import 'package:http/http.dart' as http;

Future sendRequest({String method="GET", required String route, Map? data}) async {
  dynamic response;
  String url = "http://192.168.0.104:8000/api/$route";
  try{
    switch (method) {
      case  "GET":
        response= await http.get(Uri.parse(url));
        break;
      case "POST":
        response= await http.post(Uri.parse(url), body: data);
        break;
    }
  }catch(error){
    throw error;
  }
  final jsonRes = jsonDecode(response.body);
  return jsonRes;

}

File: dashboard/src/core/config/request.js
// import axios from "axios";
// import { localStorageAction } from "./localstorage";

// axios.defaults.baseURL = "http://192.168.0.104:8000/api/";

// export const sendRequest = async ({
//   method = "GET",
//   route,
//   body,
//   includeHeaders = true,
// }) => {
//   if (!route) throw Error("URL required");

//   axios.defaults.headers.authorization = includeHeaders
//     ? `Bearer ${localStorageAction("access_token")}`
//     : "";

//   try {
//     const response = await axios.request({
//       method,
//       url: route,
//       data: body,
//     });

//     return response.data;
//   } catch (error) {
//     throw error;
//   }
// };
